packages/runtime/test/src/parser/parser.ts

```diff
@@ -428,11 +434,19 @@ describe("Parser", function () {
     });
 
     it("should return correct object for `txna`", () => {
-      const res = opcodeFromSentence(["txna", "Fee", "2"], 1, interpreter);
-      const expected = new Txna(["Fee", "2"], 1, interpreter);
+      let res = opcodeFromSentence(["txna", "Accounts", "0"], 1, interpreter);
+      let expected = new Txna(["Accounts", "0"], 1, interpreter);
+      assert.deepEqual(res, expected);
 
+      res = opcodeFromSentence(["txna", "ApplicationArgs", "2"], 1, interpreter);
+      expected = new Txna(["ApplicationArgs", "2"], 1, interpreter);
       assert.deepEqual(res, expected);
 
+      expectRuntimeError(
+        () => opcodeFromSentence(["txna", "Fee", "2"], 1, interpreter),
+        RUNTIME_ERRORS.TEAL.INVALID_OP_ARG
```

Comment:
i think this test should be in opcode-list, because this tests the field. parser only parses the sentence.

Reply:
No, this should be in parser because this validation is being done in the opcode constructor. Parser returns a new opcode, so validations in constructor should be handled in the parser tests. In `opcode-list.ts`, we are testing `op.execute()`, and not `new Op(..)`.

Reply:
ok